Fix incorrect comparison brackets in configure script

bourtemb opened the pull request:
Fix incorrect comparison brackets (#9) 
Fix search path for mysql.h file when --enable-mariadb is used

bourtemb committed:
Fix incorrect comparison brackets (#9)

bourtemb committed:
Fix search path for mysql.h file when --enable-mariadb is used

mliszcz commented on assets/m4/mysql_release.m4:
```diff
@@ -19,7 +19,7 @@ AC_DEFUN([mysql_AC_HAVE_MYSQL_VERSION],
 					MARIA=`echo $MYSQL_MICRO | cut -d '-' -f 2`
 
-					if [[ "$MARIA" == *aria* ]]; then
+					if [[[ "$MARIA" == *aria* ]]]; then
```
I assume this line will output: `if [[ "$MARIA" == *aria* ]]; then`, correct?
If yes, it will not work as `[[ ]]` is a bash builtin. `configure` script should be POSIX compatible unless you explicitly run it via bash (default shebang is `#! /bin/sh`).

I did a small test:
```bash
# test-mysql.sh
VERS=`mysql --version 2>&1 | cut -d ' ' -f 6 | cut -d ',' -f 1`
MYSQL_MICRO=`echo $VERS | cut -d '.' -f 3`
MARIA=`echo $MYSQL_MICRO | cut -d '-' -f 2`
if [[ "$MARIA" == *aria* ]]; then
  echo "Maria DB"
else
  echo "NOT Maria DB"
fi
```
```bash
$ /bin/sh test-mysql.sh
test-mysql.sh: 5: test-mysql.sh: [[: not found
NOT Maria DB
```

Maybe you tested this on a system where `sh` is symlinked to `bash`? On my Ubuntu 18.04 `sh` points to `dash`.

This should work:
`if test -z ${MARIA##*aria*}; then`

Or replace with switch:
```bash
case "$MARIA" in
  *aria*)
    echo "Maria DB"
    ;;
  *)
    echo "NOT Maria DB"
    ;;
esac
```

t-b replied:
+1 for the switch solution

bourtemb replied:
> I assume this line will output: `if [[ "$MARIA" == *aria* ]]; then`, correct?

Yes
> 
> Maybe you tested this on a system where `sh` is symlinked to `bash`? On my Ubuntu 18.04 `sh` points to `dash`.

I was probably on a system where `sh` is symlinked to `bash` indeed.
switch solution sounds fine to me too then.

mliszcz reviewed (changes requested):
I'm not sure if this will work if you run `configure` with plain sh shell, please see the inline comment.

mliszcz committed:
Use switch instead of [[ for Maria DB detection

Use switch as [[ is a bash-specific builtin not specified by POSIX
and breaks configure script when run from a plain old sh shell.

mliszcz committed:
Unify indentation in MySQL version detection script

mliszcz commented:
I think I understand why `[[` has not caused any problem even that shebang points to `/bin/sh`.

The configure script re-runs itself using some other shell (bash in my case). You can override this by setting CONFIG_SHELL, e.g.:

```
sudo CONFIG_SHELL=/bin/sh ./configure --without-zlib --with-mysql-admin=root --with-mariadbclient-include="/usr/include/mysql" --with-mariadbclient-lib="/usr/lib/x86_64-linux-gnu" --enable-dbcreate --enable-dbserver=yes
```

Now it reruns via /bin/sh:
```
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr/local'  '--without-zlib' '--with-mysql-admin=root' '--with-mariadbclient-include=/usr/include/mysql' '--with-mariadbclient-lib=/usr/lib/x86_64-linux-gnu' '--enable-dbcreate' '--enable-dbserver=yes' --cache-file=/dev/null --srcdir=.
```

And the mariadb check always fails:
```
checking for mysql... /usr/bin/mysql
checking for mysql release (at least version 5.5)... ./configure: 19576: ./configure.lineno: [[: not found
no
configure: WARNING: Not supported mysql release. Should be 5.5.x or above. Please update !
```

I do not understand why/how configure is picking bash. Per answers in [this SO question](https://stackoverflow.com/questions/161064/autoconf-using-sh-i-need-shell-bash-how-do-i-force-autoconf-to-use-bash), it might be due to the current shell being bash, but I couldn't reproduce the issue by changing only SHELL env variable.

Question: Do we still want to change the check to work in a plain sh shell?
@t-b, since Reynald is OOO, I've allowed myself to add two commits to this PR. Could you please have a look? I'm also on vacation and starting from tomorrow will have no access to my PC, so feel free to remove my commits if you prefer the original behavior.

Thanks!

t-b commented:
@mliszcz I'll have a look.

t-b commented:
@mliszcz @bourtemb Works as expected.
